motionplan/planManager.go

```diff
@@ -289,26 +293,26 @@ func (pm *planManager) planSingleAtomicWaypoint(
 		case nextSeed := <-endpointPreview:
 			return nextSeed.Q(), &resultPromise{future: solutionChan}, nil
 		case planReturn := <-solutionChan:
-			if planReturn.planerr != nil {
-				return nil, nil, planReturn.planerr
+			if planReturn.err != nil {
+				return nil, nil, planReturn.err
 			}
-			steps := nodesToInputs(planReturn.steps)
-			return steps[len(steps)-1], &resultPromise{steps: steps}, nil
+			seed := planReturn.steps[len(planReturn.steps)-1].Q()
+			return seed, &resultPromise{steps: planReturn.steps}, nil
 		}
 	} else {
 		// This ctx is used exclusively for the running of the new planner and timing it out. It may be different from the main `ctx`
 		// timeout due to planner fallbacks.
 		plannerctx, cancel := context.WithTimeout(ctx, time.Duration(pathPlanner.opt().Timeout*float64(time.Second)))
 		defer cancel()
-		steps, err := pathPlanner.plan(plannerctx, goal, seed)
+		nodes, err := pathPlanner.plan(plannerctx, goal, seed)
 		if err != nil {
 			return nil, nil, err
 		}
 
-		smoothedPath := nodesToInputs(pathPlanner.smoothPath(ctx, steps))
+		smoothedPath := pathPlanner.smoothPath(ctx, nodes)
 
 		// Update seed for the next waypoint to be the final configuration of this waypoint
-		seed = smoothedPath[len(smoothedPath)-1]
+		seed = smoothedPath[len(smoothedPath)-1].Q()
```

Comment:
Is this fixing a bug?

Reply:
This is 1:1, the type of `smoothedPath` changed so the `.Q()` is now necessary to make `seed` be the same type